### PR TITLE
Fixes a race condition in client events monitoring

### PR DESCRIPTION
**Pull request by cpuguy83**
In cases where there is high latency (ie, not-local network)
`waitExitOrRemoved` was not receiving events for short-lived containers.
This caused the client to hang while waiting for a notification that the
container has stopped.

This happens because `client.Events()` returns immediately and spins a
goroutine up to process events. The problem here is it returns before
the request to the events endpoint is even made.
Even without high-latency issues, there is no guarantee that the
goroutine is even scheduled by the time the function returns.

### CODE REVIEW

File: cli/command/container/utils.go [hunk 1]
```diff
@@ -92,6 +94,7 @@ func waitExitOrRemoved(ctx context.Context, dockerCli *command.DockerCli, contai
 			}
 		}
 	}()
+	<-started
```
Comment (tonistiigi) [inline cli/command/container/utils.go]: I get the change in `client/events.go` but not sure I understand the change in this file. In any case, this seems best effort as it doesn't guarantee any particular execution order.

**Comment by tonistiigi** [comment]
cc @WeiZhang555

File: cli/command/container/utils.go (hunk 1 above)
Comment (cpuguy83) [inline cli/command/container/utils.go]: It just makes sure that at least the goroutine is running before it returns... which then will make a call to start the container.

Comment (WeiZhang555) [inline cli/command/container/utils.go]: Hmmm, I see your point, but I don't think it's that necessary, though it doesn't hurt anything either.

Even function returns before goroutine is scheduled, at least the `Event` call is already made, so we could monitor the container correctly.  This goroutine only influences outputting data to `statusChan`.

Comment (cpuguy83) [inline cli/command/container/utils.go]: You're right, it's not really helping anything, I'll remove.

**Comment by aaronlehmann** [comment]
LGTM

**Comment by tonistiigi** [comment]
LGTM

**Pull request merged**